great_expectations_provider/operators/validate_checkpoint.py

```diff
@@ -40,15 +43,54 @@ def execute(self, context: Context) -> CheckpointDescriptionDict:
         import great_expectations as gx
 
         gx_context: AbstractDataContext
+        file_context_generator: Generator[FileDataContext, None, None] | None = None
 
         if self.context_type == "file":
             if not self.configure_file_data_context:
                 raise ValueError(
                     "Parameter `configure_file_data_context` must be specified if `context_type` is `file`"
                 )
-            gx_context = self.configure_file_data_context()
+            elif inspect.isgeneratorfunction(self.configure_file_data_context):
+                file_context_generator = self.configure_file_data_context()
+                gx_context = self._get_value_from_generator(file_context_generator)
+            else:
+                file_context_fn = cast(
+                    Callable[[], FileDataContext], self.configure_file_data_context
+                )
+                gx_context = file_context_fn()
         else:
             gx_context = gx.get_context(mode=self.context_type)
         checkpoint = self.configure_checkpoint(gx_context)
         result = checkpoint.run(batch_parameters=self.batch_parameters)
+
+        if file_context_generator:
+            self._allow_generator_teardown(file_context_generator)
+
         return result.describe_dict()
+
+    def _get_value_from_generator(
+        self, generator: Generator[FileDataContext, None, None]
+    ) -> FileDataContext:
+        try:
+            return next(generator)
+        except StopIteration:
+            raise RuntimeError("Generator must yield exactly once; did not yield")
+
+    def _allow_generator_teardown(self, generator: Generator) -> None:
+        """Run the generator to completion to allow for any cleanup/teardown.
+
+        Also does some error handling to ensure the generator doesn't yield more than once.
+        """
+        try:
+            # Check if we have another yield (this is an error case)
+            next(generator)
+        except StopIteration:
+            pass
+        else:
+            # we had an extra yield; we'll raise, but first, let's run out the generator
+            # so it does any teardown/cleanup
+            for _ in generator:
```

Comment:
I think there's a tradeoff here. If there's an infinite number of yields, this will hang. But if there are a finite amount and then some teardown, this will ensure the teardown. I think I lean toward removing this though. If you do something wrong, don't expect teardown. Thoughts?

Reply:
I decided to remove this code. If you are misconfiguring your generator, I don't think we should make assumptions, and we don't need to clean up. Looks like pytest follows the same philosophy.